### PR TITLE
simplify the argument parsing

### PR DESCRIPTION
**Pull request by goneri**
- use argparse's subparser to handle the different actions
- let argparse handle the parsing/validation of virt_lightning_yaml
- simplify a bit the logic of Configuration

### CODE REVIEW

File: virt_lightning/shell.py [hunk 1]
```diff
@@ -169,29 +145,25 @@ def iconify(v):
...
-                "name": status["name"],
-                "ipv4": status["ipv4"] or "waiting",
-                "context": status["context"],
-                "username": status["username"],
-                "ssh_ping": iconify(status["ssh_ping"]),
-            }
+    for status in get_status(hv, context):
+        results[status["name"]] = {
+            "name": status["name"],
+            "ipv4": status["ipv4"] or "waiting",
```
Comment (zloidemon) [inline virt_lightning/shell.py]: You can change that code to `status.get("ipv4", "waiting")`

File: virt_lightning/shell.py [hunk 2]
```diff
@@ -215,87 +187,102 @@ def storage_dir(configuration):
...
-    elif args.action == "status_live":
-        status(configuration, args.context, True)
-    elif args.action == "storage_dir":
-        storage_dir(configuration)
+
+    configuration = Configuration()
+    if args.config:
+        configuration.load_fd(args.config)
+
+    globals()[args.action](configuration=configuration, **vars(args))
```
Comment (zloidemon) [inline virt_lightning/shell.py]: Are you sure about using `globals`?

File: virt_lightning/shell.py (hunk 1 above)
Comment (goneri) [reply virt_lightning/shell.py]: With the current code, `status["ipv4"]` can be `None`, `get("ipv4", "foo")` would return `None`, not `foo`.

File: virt_lightning/shell.py (hunk 2 above)
Comment (goneri) [reply virt_lightning/shell.py]: The alternative is to maintain a dict() with a name/function mapping.

**Pull request merged**